enhancements/olm/olmv1-webhook-support.md

```diff
@@ -0,0 +1,380 @@
+there should be any specific Hypershift considerations. I.e. no resources will be installed in the central cluster.
+
+#### Standalone Clusters
+
+This change only affects the types of workloads that the customer will run. It should not have any major effect on 
+OLMv1 itself.
+
+#### Single-node Deployments or MicroShift
+
+This change only affects the types of workloads that the customer will run. It should not have any major effect on
+OLMv1 itself.
+
+### Implementation Details/Notes/Constraints
+
+The ClusterServiceVersion contained in the registry+v1 bundle to be installed is transpiled into the final set of
+manifests that will be applied to the cluster (i.e. CRDs, Deployments, Services, etc.). The ClusterServiceVersion
+can contain definitions for Validating, Mutating, and Conversion Webhooks, which form the basis for:
+- MutatingWebhookConfiguration resource(s)
+- ValidatingWebhookConfiguration resource(s)
+- Service resource(s) that back the webhooks
+- .spec.Conversion for any CRD resource with a conversion webhook
+
+The relevant generated resources, i.e. CRDs, Services, Mutating-/ValidatingWebhookConfigurations,
+will carry the appropriate Openshift-ServiceCA annotations for the generation and management TLS certificates for the
+webhook service(s). In addition, any Deployment resources that host the business logic for the webhook(s) will be 
+configured with the correct volumes and volume mounts to consume the generated TLS certificate.
+
+### Risks and Mitigations
+
+Because we are transpiling a ClusterServiceVersion into a set of coherent resources, there is a security implication that
+well crafted bundles _could_ potentially be malicious in nature and could be used to subvert to host cluster. These
+risks are mitigated by:
+- Allowing cluster admins to configure the scope of RBAC they are comfortable with for a particular bundle
+- Allowing cluster admins to control who gets write access to ClusterExtensions/ClusterCatalogs
+- Allowing cluster admins to select the ClusterCatalogs from which content can be sourced
+
+Webhooks come with services, which could be a potential avenue for data leaks or security issues. All
+communications between webhook components is secured via TLS certificates generated and rotated automatically by
+Openshift-ServiceCA.
+
+A faulty webhook can have a rather large blast radius. Registry+v1 bundles can contain validating and mutating webhooks
+for, in principle, any resource. To limit the blast radius a bit, the following boundaries are put around the support
+for webhooks:
+
+- The validating/mutating webhooks configuration cannot contain rules referencing:
+  * the api groups: '*' or 'olm.operatorframework.io'
+  * the api group 'admissionregistration.k8s.io' and any combination of resources: '*', 'validatingwebhookconfigurations', and 'mutatingwebhookconfigurations'
+- Conversion webhooks can only be configured for CRDs owned by the bundle
+
+
+### Drawbacks
+
+None that I could think of.
```

Comment:
Wouldn't Service CA requirement lock OLMv1 to OpenShift and prevent it from being used in vanilla k8s?

Reply:
Upstream OLM uses cert-manager. Do we need to call out upstream concerns in the enhancements? Or is the focus more on OpenShift?